src/test/java/com/yourcompany/utils/SauceUtils.java

```diff
@@ -25,60 +29,56 @@ public static void UpdateResults(String username, String accessKey, boolean test
             throws JSONException, IOException {
         SauceREST client = getSauceRestClient(username, accessKey);
         Map<String, Object> updates = new HashMap<String, Object>();
-        addBuildNumberToUpdate(updates);
         updates.put("passed", testResults);
         client.updateJobInfo(sessionId, updates);
     }
+
+    public static DesiredCapabilities createCapabilities(String value) throws FileNotFoundException {
+        FileReader file = new FileReader("src/test/java/com/yourcompany/utils/platforms.yml");
+        Map<String, Object> platforms = (Map<String, Object>) new Yaml().load(file);
+        Map<String, Object> platform = (Map<String, Object>) platforms.get(value);
+        DesiredCapabilities capabilities = new DesiredCapabilities();
+        for (String key : platform.keySet()) {
+            capabilities.setCapability(key, platform.get(key));
+        }
+        return capabilities;
+    }
+
     /**
      * Populates the <code>updates</code> map with the value of the system property/environment variable
      * with the following name:
      * <ol>
-     *     <li>SAUCE_BAMBOO_BUILDNUMBER</li>
-     *     <li>JENKINS_BUILD_NUMBER</li>
-     *     <li>BUILD_TAG</li>
-     *     <li>BUILD_NUMBER</li>
-     *     <li>TRAVIS_BUILD_NUMBER</li>
-     *     <li>CIRCLE_BUILD_NUM</li>
+     * <li>SAUCE_BAMBOO_BUILDNUMBER</li>
+     * <li>JENKINS_BUILD_NUMBER</li>
+     * <li>BUILD_TAG</li>
+     * <li>BUILD_NUMBER</li>
+     * <li>TRAVIS_BUILD_NUMBER</li>
+     * <li>CIRCLE_BUILD_NUM</li>
      * </ol>
-     * @param updates String,Object pair containing job updates
      */
-    private static void addBuildNumberToUpdate(Map<String, Object> updates) {
-        //try Bamboo
-        String buildNumber = readPropertyOrEnv("SAUCE_BAMBOO_BUILDNUMBER", null);
-        if (buildNumber == null || buildNumber.equals("")) {
-            //try Jenkins
-            buildNumber = readPropertyOrEnv("JENKINS_BUILD_NUMBER", null);
-        }
-
-        if (buildNumber == null || buildNumber.equals("")) {
-            //try BUILD_TAG
-            buildNumber = readPropertyOrEnv("BUILD_TAG", null);
-        }
+    public static String getBuildName() {
```

Comment:
We can't possibly cover every possible environment variable for every CI tool, so we shouldn't try to.  We should adopt a single environment variable and treat it as the interface.  Since we've been using BUILD_TAG, I vote we stick with that.  And the each tool can adapt to that or use the Sauce OnDemand plugin.

Also, JENKINS_BUILD_NUMBER is not unique (may also be true for some of these other variables).  JENKINS_BUILD_TAG is the same as JOB_NAME + BUILD_NUMBER

e.g.
export BUILD_TAG="my job name" + TRAVIS_BUILD_NUMBER

If we do want to support this, it should be abstracted into an adapter class.